hadoop-common-project/hadoop-common/pom.xml

```diff
@@ -669,6 +676,36 @@
           </filesets>
         </configuration>
       </plugin>
+        <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-shade-plugin</artifactId>
+            <version>${maven-shade-plugin.version}</version>
+            <executions>
+                <execution>
+                    <phase>package</phase>
+                    <goals>
+                        <goal>shade</goal>
+                    </goals>
+                    <configuration>
+                        <artifactSet>
+                            <includes>
+                                <include>io.netty:netty-all:jar:*</include>
+                            </includes>
+                            <excludes>
+                                <!-- this is netty 3 -->
+                                <exclude>*:netty</exclude>
+                            </excludes>
+                        </artifactSet>
+                        <relocations>
+                            <relocation>
+                                <pattern>io.netty</pattern>
+                                <shadedPattern>hrpc.io.netty</shadedPattern>
```

Comment:
This relocation path will break HBase (it has classpath checks forbidding unfamiliar prefixes)
Can we change it to org.apache.hadoop.io.netty?
In fact, it would be a good idea to shade netty into hadoop-thirdparty (https://github.com/apache/hadoop-thirdparty)which relocates to org.apache.hadoop.thirdparty.*

Reply:
Changed it to

`<relocation>
    <pattern>io.netty</pattern>
    <shadedPattern>org.apache.hadoop.thirdparty.io.netty</shadedPattern>
</relocation>`